repository_service_tuf_worker/repository.py

```diff
@@ -1425,97 +1424,84 @@ def sign_metadata(
     ) -> Dict[str, Any]:
         """Add signature to metadata for pending signing event.
 
-        Add signature (from payload) to cached role metadata (from settings)
-        for the role that matches the passed rolename (from payload), if a
-        signing event exists for that role, and the signature is valid.
-
-        If the signature threshold is reached, the signing event is finalized.
+        Add signature (from payload) to cached root metadata (from settings),
+        if a signing event exists, and the signature is valid.
 
-        ** Signing event types (and details) **
+        Signing event types are 'bootstrap' or 'metadata update'.
 
-        BOOTSTRAP: Only root metadata can be updated in this event. To verify
-        the passed signature, the keys and threshold are read from the root
-        metadata to be updated itself. If the threshold is reached, the
-        bootstrap process is finalized.
-
-        METADATA UPDATE: Root, targets or delegated targets metadata can be
-        updated in this event. Depending on the metadata type, the authorized
-        public keys and threshold are read from different delegating metadata.
+        If the signature threshold is reached, the signing event is finalized,
+        otherwise it remains in pending state.
         """
-        rolename = payload["role"]
-        signature_dict = payload["signature"]
 
-        # Assert pending signing event
-        metadata_dict = self._settings.get_fresh(f"{rolename.upper()}_SIGNING")
+        def _result(status, error=None, bootstrap=None, update=None):
+            details = {}
+            if status:
+                details["message"] = "Signature Processed"
+            else:
+                details["message"] = "Signature Failed"
+            if error:
+                details["error"] = error
+            if bootstrap:
+                details["bootstrap"] = bootstrap
+            if update:
```

Comment:
You wouldn't call `bootstrap` and `update` and the same time.
So maybe it's better to remove a possibility of a mistake, make it:
```suggestion
            if bootstrap:
                details["bootstrap"] = bootstrap
            elif update:
```

Reply:
The `elif` alone does not remove the possibility of mistake. I'd have to add a `raise` on `else`, which seems overkill for a nested function, whose calls I control and test. But yeah, the elif removes an unnecessary evaluation. I can change this. 👍